translate/storage/pypo.py

```diff
@@ -757,20 +760,23 @@ class pofile(pocommon.pofile):
     """A .po file containing various units"""
     UnitClass = pounit
 
-    def parse(self, input):
+    def parse(self, input, duplicatestyle=None):
         """Parses the given file or file source string."""
-        if True:
-            if hasattr(input, 'name'):
-                self.filename = input.name
-            elif not getattr(self, 'filename', ''):
-                self.filename = ''
-            if isinstance(input, str):
-                input = BytesIO(input)
-            # clear units to get rid of automatically generated headers before parsing
-            self.units = []
-            poparser.parse_units(poparser.ParseState(input, pounit), self)
-
-    def removeduplicates(self, duplicatestyle="merge"):
+        if hasattr(input, 'name'):
+            self.filename = input.name
+        elif not getattr(self, 'filename', ''):
+            self.filename = ''
+        if isinstance(input, str):
+            input = BytesIO(input)
+        # clear units to get rid of automatically generated headers before parsing
+        self.units = []
+        poparser.parse_units(poparser.ParseState(input, pounit), self)
+
+        # duplicates are now removed by default unless duplicatestyle=allow
+        if duplicatestyle != "allow":
```

Comment:
im not really following

we cant have duplicatestyle="allow" as the default in parse or it wouldnt fix the problem - it would revert to old behaviour